core/src/test/scala/org/apache/spark/storage/ShuffleBlockFetcherIteratorSuite.scala

```diff
@@ -104,10 +113,13 @@ class ShuffleBlockFetcherIteratorSuite extends SparkFunSuite {
       assert(subIterator.isSuccess,
         s"iterator should have 5 elements defined but actually has $i elements")
 
-      // Make sure we release the buffer once the iterator is exhausted.
+      // Make sure we release buffers when a wrapped input stream is closed.
       val mockBuf = localBlocks.getOrElse(blockId, remoteBlocks(blockId))
+      val wrappedInputStream = new BufferReleasingInputStream(mock(classOf[InputStream]), iterator)
```

Comment:
Why not use the actual input stream here (I think this is subIterator, above, which should probably be renamed to inputStream)?